src/clickup/api.ts

```diff
@@ -0,0 +1,101 @@
+import {IClickUpConfig,
+   IClickUpQueryParams, 
+  IClickUpTeamResponse,
+  IClickUpSpaceResponse,
+  IClickUpFolderResponse,
+  IClickUpListResponse,
+  IClickUpTaskResponse,
+  IClickUpTask,
+} from './types';
+import { HttpClient } from '../common/HttpClient';
+import { IRequestConfig } from '../common/types';
+import { CLICKUP_PATHS } from "./consts";
+
+export class ClickUpApi extends HttpClient {
```

Comment:
lets use https.Agent and set max socket to 10, provide a way to configure via .env

Reply:
i kinda don't see much difference
Execution Time: 73.71431400000006 ms with 1 sockets
Execution Time: 50.69687899999985 ms with 1 sockets
Execution Time: 52.47996400000011 ms with 1 sockets
Execution Time: 51.51298299999985 ms with 1 sockets
Execution Time: 50.511122000000114 ms with 1 sockets

Execution Time: 82.42646299999979 ms with 10 sockets
Execution Time: 73.431959 ms with 10 sockets
Execution Time: 72.74709800000005 ms with 10 sockets
Execution Time: 81.47947299999987 ms with 10 sockets
Execution Time: 56.60399899999993 ms with 10 sockets